axonserver/src/main/java/io/axoniq/axonserver/grpc/EventStoreService.java

```diff
@@ -180,8 +181,31 @@ public void listAggregateEvents(GetAggregateEventsRequest request,
     }
 
 
-    public StreamObserver<GetEventsRequest> listEvents(StreamObserver<InputStream> responseObserver) {
-        return eventDispatcher.listEvents(contextProvider.getContext(), authenticationProvider.get(), responseObserver);
+    public StreamObserver<GetEventsRequest> listEvents(StreamObserver<SerializedEventWithToken> responseObserver) {
+        String context = contextProvider.getContext();
+        Executor executor = grpcFlowControlExecutorProvider.provide();
+        OutgoingStream<SerializedEventWithToken> outgoingStream = new FlowControlledOutgoingStream<>((CallStreamObserver<SerializedEventWithToken>) responseObserver,
+                                                                                                     executor);
+        Sinks.Many<GetEventsRequest> requestFlux = Sinks.many()
+                                                        .unicast()
+                                                        .onBackpressureBuffer();
+        outgoingStream.accept(eventDispatcher.events(context, authenticationProvider.get(), requestFlux.asFlux()));
+        return new StreamObserver<GetEventsRequest>() {
+            @Override
+            public void onNext(GetEventsRequest getEventsRequest) {
+                requestFlux.tryEmitNext(getEventsRequest);
```

Comment:
error handling

Reply:
Just a thought, can we postpone the call to eventDispatcher.events to the point where we have received the first GetEventsRequest, and pass the initial request as a parameter?

Reply:
I guess we can, but I'm not sure what is the real benefit of this? Do you think that the initial request is going to be delayed that much that it might influence the resources `outgoingStream.accept` reserves?

Reply:
Actually, it makes a lot of sense to invoke `eventDispatcher.events` method only when someone sends the first request. Will change it.